R/endpoints.R

```diff
@@ -119,6 +119,39 @@ model_status <- function(queue) {
   }
 }
 
+model_result <- function(queue) {
+  function(id) {
+    task_status <- queue$queue$task_status(id)
+    if (task_status == "COMPLETE") {
+      hintr:::process_result(queue$result(id))
+    } else if (task_status == "ERROR") {
+      result <- queue$result(id)
+      trace <- c(sprintf("# %s", id), result$trace)
+      error_data <- structure(result$message, trace = trace)
+      pkgapi::pkgapi_stop(error_data, "MODEL_RUN_FAILED")
+    } else if (task_status == "ORPHAN") {
+      pkgapi::pkgapi_stop(tr_("MODEL_RESULT_CRASH"), "MODEL_RUN_FAILED")
+    } else if (task_status == "INTERRUPTED") {
+      pkgapi::pkgapi_stop(tr_("MODEL_RUN_CANCELLED"), "MODEL_RUN_FAILED")
+    } else { # ~= MISSING, PENDING, RUNNING
+      pkgapi::pkgapi_stop(tr_("MODEL_RESULT_MISSING"),
+                          "FAILED_TO_RETRIEVE_RESULT")
+    }
+  }
+}
+
+model_cancel <- function(queue) {
+  function(id) {
+    tryCatch({
+      queue$cancel(id)
+      json_null()
+    },
+    error = function(e) {
+      pkgapi::pkgapi_stop(e$message, "FAILED_TO_CANCEL")
```

Comment:
Do we know what could cause this to fail? We'll end up with English coming through here always, but that's unavoidable if the errors come from rrq

Reply:
I think only concrete example is trying to cancel something where the key doesn't exist. Though I think unlikely something will go wrong.